watchtower-plugin/src/retrier.rs

```diff
@@ -16,174 +16,248 @@ use crate::wt_client::WTClient;
 
 pub struct RetryManager {
     wt_client: Arc<Mutex<WTClient>>,
-    retriers: Arc<Mutex<HashMap<TowerId, Retrier>>>,
+    unreachable_towers: UnboundedReceiver<(TowerId, Locator)>,
+    max_elapsed_time_secs: u16,
+    max_interval_time_secs: u16,
+    retriers: HashMap<TowerId, Retrier>,
 }
 
 impl RetryManager {
-    pub fn new(wt_client: Arc<Mutex<WTClient>>) -> Self {
+    pub fn new(
+        wt_client: Arc<Mutex<WTClient>>,
+        unreachable_towers: UnboundedReceiver<(TowerId, Locator)>,
+        max_elapsed_time_secs: u16,
+        max_interval_time_secs: u16,
+    ) -> Self {
         RetryManager {
             wt_client,
-            retriers: Arc::new(Mutex::new(HashMap::new())),
+            unreachable_towers,
+            max_elapsed_time_secs,
+            max_interval_time_secs,
+            retriers: HashMap::new(),
         }
     }
-    pub async fn manage_retry(
-        &mut self,
-        max_elapsed_time_secs: u16,
-        max_interval_time_secs: u16,
-        mut unreachable_towers: UnboundedReceiver<(TowerId, Locator)>,
-    ) {
+
+    /// Starts the retrier manager's main logic loop.
+    /// This method will keep running until the `unreachable_towers` sender disconnects.
+    ///
+    /// It will receive any `(tower_id, locator)` pair and try to send the appointment with `locator` to
+    /// the tower with `tower_id`. This is done by spawning a tokio thread for each `tower_id` that tries
+    /// to send all the pending appointments.
+    pub async fn manage_retry(&mut self) {
         log::info!("Starting retry manager");
 
         loop {
-            let (tower_id, locator) = unreachable_towers.recv().await.unwrap();
-            // Not start a retry if the tower is flagged to be abandoned
-            {
-                let wt_client = self.wt_client.lock().unwrap();
-                if !wt_client.towers.contains_key(&tower_id) {
-                    log::info!("Skipping retrying abandoned tower {}", tower_id);
-                    continue;
+            match self.unreachable_towers.try_recv() {
+                Ok((tower_id, locator)) => {
+                    // Not start a retry if the tower is flagged to be abandoned
+                    if !self
+                        .wt_client
+                        .lock()
+                        .unwrap()
+                        .towers
+                        .contains_key(&tower_id)
+                    {
+                        log::info!("Skipping retrying abandoned tower {}", tower_id);
+                        continue;
+                    }
+                    self.add_pending_appointment(tower_id, locator);
```

Comment:
> You could also add an additional state to the RetrierStatus enum like Ready for freshly initialized retriers and check whether the retrier is running or ready. Currently the retriers start in Success mode, which is harmless but a bit misleading

Also one other solution would to rename success and failure to more generic terms (like good/bad). This would clear the confusion if we state what these terms mean (e.g. good = this retrier can be used, bad = get rid of me right away!!).